JSON schema file for YAML configuration schema validation

buddhikac96 opened the pull request:
Fixes https://github.com/winsw/winsw/issues/537

oleg-nenashev commented on doc/yamlConfigFile.md:
```diff
@@ -28,6 +28,13 @@ log:
+## YAML configuration schema validation
+
+Users can validate YAML configurations file against JSON schema.
+First you should convert YAML file into JSON and then you can
```
It is not needed if you use https://marketplace.visualstudio.com/items?itemName=redhat.vscode-yaml . I would suggest documenting such use-case which is more friendly to users

oleg-nenashev commented on doc/yamlConfigurationSchema.json:
```diff
@@ -0,0 +1,365 @@
+  "$schema": "http://json-schema.org/draft-07/schema",
+  "type": "object",
+  "title": "Windows Service Wrapper. YAML configuration schema",
+  "description": "This JSON schema validate the YAML configuaration file.",
```
Link the YAML config documentation page for more details?

oleg-nenashev replied:
```suggestion
  "description": "This JSON schema validate the YAML configuration file.",
```

oleg-nenashev reviewed (changes requested):
I have tested it with https://marketplace.visualstudio.com/items?itemName=redhat.vscode-yaml , and it works well. There is an issue with boolean properties which are registered as String with boolPattern in the schema. Would be great to actually make them boolean if possible, but it is a subject for a separate discussion with @buddhikac96 and @NextTurn . Might be fine as well

Requesting changes because of the typo

oleg-nenashev reviewed (approved):
Fine with me, thanks for addressing the comments.
We can definitely start from that and improve in follow-ups

oleg-nenashev commented on doc/yamlConfigurationSchema.json:
```diff
@@ -0,0 +1,365 @@
+      }
+    }
+  }
+}
```
```suggestion
}

```

buddhikac96 commented:
> Fine with me, thanks for addressing the comments.
> We can definitely start from that and improve in follow-ups

Thanks. Are we willing to integrate YAML schema validation someday or we will keep it to use from VSCode?

NextTurn commented on doc/yamlConfigFile.md:
```diff
@@ -28,6 +28,13 @@ log:
+Users can validate YAML configurations file against JSON schema.
+First you should convert YAML file into JSON and then you can
+validate the JSON file against JSON schema.
+[Find the JSON schema in this link.](yamlConfigurationSchema.json)
```
After this link was removed the JSON file is not referenced anywhere.

buddhikac96 replied:
So how should I reference the file? Just use the file name without a link?
